In [1]:
import numpy as np
import pandas as pd

In [3]:
hotels_df = pd.read_csv(r'rough_data\515k_hotel_reviews_in_europe\hotel_reviews.csv')

In [4]:
hotels_df

Hotel_Address  \
0        s Gravesandestraat 55 Oost 1092 AA Amsterdam ...   
1        s Gravesandestraat 55 Oost 1092 AA Amsterdam ...   
2        s Gravesandestraat 55 Oost 1092 AA Amsterdam ...   
3        s Gravesandestraat 55 Oost 1092 AA Amsterdam ...   
4        s Gravesandestraat 55 Oost 1092 AA Amsterdam ...   
...                                                   ...   
515733  Wurzbachgasse 21 15 Rudolfsheim F nfhaus 1150 ...   
515734  Wurzbachgasse 21 15 Rudolfsheim F nfhaus 1150 ...   
515735  Wurzbachgasse 21 15 Rudolfsheim F nfhaus 1150 ...   
515736  Wurzbachgasse 21 15 Rudolfsheim F nfhaus 1150 ...   
515737  Wurzbachgasse 21 15 Rudolfsheim F nfhaus 1150 ...   

        Additional_Number_of_Scoring Review_Date  Average_Score  \
0                                194    8/3/2017            7.7   
1                                194    8/3/2017            7.7   
2                                194   7/31/2017            7.7   
3                                194   7/31/2017            7.7   
4                                194   7/24/2017            7.7   
...                              ...         ...            ...   
515733                           168   8/30/2015            8.1   
515734                           168   8/22/2015            8.1   
515735                           168   8/19/2015            8.1   
515736                           168   8/17/2015            8.1   
515737                           168    8/9/2015            8.1   

                   Hotel_Name Reviewer_Nationality  \
0                 Hotel Arena              Russia    
1                 Hotel Arena             Ireland    
2                 Hotel Arena           Australia    
3                 Hotel Arena      United Kingdom    
4                 Hotel Arena         New Zealand    
...                       ...                  ...   
515733  Atlantis Hotel Vienna              Kuwait    
515734  Atlantis Hotel Vienna             Estonia    
515735  Atlantis Hotel Vienna               Egypt    
515736  Atlantis Hotel Vienna              Mexico    
515737  Atlantis Hotel Vienna             Hungary    

                                          Negative_Review  \
0        I am so angry that i made this post available...   
1                                             No Negative   
2        Rooms are nice but for elderly a bit difficul...   
3        My room was dirty and I was afraid to walk ba...   
4        You When I booked with your company on line y...   
...                                                   ...   
515733   no trolly or staff to help you take the lugga...   
515734           The hotel looks like 3 but surely not 4    
515735   The ac was useless It was a hot week in vienn...   
515736                                        No Negative   
515737       I was in 3rd floor It didn t work Free Wife    

        Review_Total_Negative_Word_Counts  Total_Number_of_Reviews  \
0                                     397                     1403   
1                                       0                     1403   
2                                      42                     1403   
3                                     210                     1403   
4                                     140                     1403   
...                                   ...                      ...   
515733                                 14                     2823   
515734                                 11                     2823   
515735                                 19                     2823   
515736                                  0                     2823   
515737                                 13                     2823   

                                          Positive_Review  \
0        Only the park outside of the hotel was beauti...   
1        No real complaints the hotel was great great ...   
2        Location was good and staff were ok It is cut...   
3        Great location in nice surroun

In [5]:
hotels_df['Hotel_Address'].value_counts()

163 Marsh Wall Docklands Tower Hamlets London E14 9SJ United Kingdom              4789
372 Strand Westminster Borough London WC2R 0JJ United Kingdom                     4256
Westminster Bridge Road Lambeth London SE1 7UT United Kingdom                     4169
Scarsdale Place Kensington Kensington and Chelsea London W8 5SY United Kingdom    3578
7 Pepys Street City of London London EC3N 4AF United Kingdom                      3212
                                                                                  ... 
2 4 Boundary Street Hackney London E2 7DD United Kingdom                            12
Filmteichstra e 5 10 Favoriten 1100 Vienna Austria                                  12
13 Rue Fran ois Ory 92120 Paris France                                              10
Via M Buonarroti 13 Fiera Milano City 20149 Milan Italy                             10
Johann Staud Stra e 32 16 Ottakring 1160 Vienna Austria                              8
Name: Hotel_Address, Length: 1493, dtype: i

In [6]:
hotels_df['Hotel_Name'].value_counts()

Britannia International Hotel Canary Wharf           4789
Strand Palace Hotel                                  4256
Park Plaza Westminster Bridge London                 4169
Copthorne Tara Hotel London Kensington               3578
DoubleTree by Hilton Hotel London Tower of London    3212
                                                     ... 
Hotel Daniel Paris                                     12
Le Lavoisier                                           12
Mercure Paris Porte d Orleans                          10
Hotel Wagner                                           10
Hotel Gallitzinberg                                     8
Name: Hotel_Name, Length: 1492, dtype: int64

Number of unique Hotel_Name values only differs from number of uniqe Hotel_address values by 1 over the course of 1492 hotels, so I can use Hotel_Name as a unique identifier. 

There are 1492 unique hotels in the dataset, with a total of 515,738 reviews.

In [27]:
hotels_df.groupby('Hotel_Name')['Reviewer_Score'].agg(['mean','count']).sort_values('count', ascending=False)[:1081]
                                                                                                             

mean  count
Hotel_Name                                                        
Britannia International Hotel Canary Wharf         6.826644   4789
Strand Palace Hotel                                8.128853   4256
Park Plaza Westminster Bridge London               8.652866   4169
Copthorne Tara Hotel London Kensington             8.094243   3578
DoubleTree by Hilton Hotel London Tower of London  8.662765   3212
...                                                     ...    ...
Bassano                                            8.131000    100
Hotel Zeitgeist Vienna Hauptbahnhof                8.559000    100
The Berkeley                                       9.064000    100
Best Western Aulivia Op ra                         7.844000    100
The Portobello Hotel                               9.032323     99

[1081 rows x 2 columns]

1080 Unique hotels have 100+ reviews.

In [34]:
hotels_df.groupby('Hotel_Name')['Reviewer_Score'].agg(['mean','count']).sort_values('count', ascending=False)[:1335]

mean  count
Hotel_Name                                                        
Britannia International Hotel Canary Wharf         6.826644   4789
Strand Palace Hotel                                8.128853   4256
Park Plaza Westminster Bridge London               8.652866   4169
Copthorne Tara Hotel London Kensington             8.094243   3578
DoubleTree by Hilton Hotel London Tower of London  8.662765   3212
...                                                     ...    ...
Melia Paris Vendome                                8.298039     51
Hotel Dieci                                        7.740000     50
H tel des Academies et des Arts                    8.896000     50
Ilunion Bel Art                                    8.134000     50
Roomz Vienna                                       8.518367     49

[1335 rows x 2 columns]

1,334 Unique hotels have 50+ reviews

In [40]:
hotels_df.groupby('Hotel_Name')['Reviewer_Score'].agg(['mean','count']).sort_values('count', ascending=False)[:1446]

mean  count
Hotel_Name                                                        
Britannia International Hotel Canary Wharf         6.826644   4789
Strand Palace Hotel                                8.128853   4256
Park Plaza Westminster Bridge London               8.652866   4169
Copthorne Tara Hotel London Kensington             8.094243   3578
DoubleTree by Hilton Hotel London Tower of London  8.662765   3212
...                                                     ...    ...
Hotel G tico                                       8.092000     25
Mercure Paris 15 Porte de Versailles               7.828000     25
Hotel Tour d Auvergne Opera                        8.476000     25
Hotel Villa Lafayette Paris IX                     9.228000     25
Goralska R sidences H tel Paris Bastille           9.458333     24

[1446 rows x 2 columns]

1,445 Unique hotels have 25+ reviews. This is nearly the whole dataset. Likely this dataset had some kind of sort or minimum criteria by reviews / property. Or maybe all luxury hotels have been reviewed at least tens of times.

In [41]:
hotels_df['Reviewer_Score'].mean()

8.395076569886239

In [44]:
#Scale adjusts this 10-point system down to a more common 5-point ('star') system.
scale_adjustment = 5/10

adjusted_mean = hotels_df['Reviewer_Score'].mean() * scale_adjustment
adjusted_mean

4.1975382849431195